### PR TITLE
typescript(vx-voronoi): re-write package in TypeScript

### LINKED ISSUE #511: Re-write `@vx/voronoi` in TypeScript

### PR DESCRIPTION
**Pull request by williaster**
#### :rocket: Enhancements

This PR builds off #488 which introduces Typescript build config, and re-writes the `@vx/voronoi` package in TypeScript. 

It also removes an unneeded `@vx/group` dependency. Closes #511.

#### :boom: Breaking Changes
- This PR introduces `React.Fragment`s, which requires bumping the `peerDep` for `react` to `^16.3.0-0`

**Tests**
- [x] CI
- [x] functional `/voronoi` demo

@hshoff @schillerk @milesj @kristw @rudeg

### CODE REVIEW

File: packages/vx-voronoi/src/components/VoronoiPolygon.tsx [hunk 1]
```diff
@@ -0,0 +1,23 @@
+import React from 'react';
+import cx from 'classnames';
+
+export type VoronoiPolygonProps = {
+  /** Override render function which is provided polygon and generated path. */
+  children?: ({ path, polygon }: { path: string; polygon: number[][] }) => React.ReactNode;
+  /** className to apply to path element. */
+  className?: string;
+  /** Array of coordinate arrays for the polygon (e.g., [[x,y], [x1,y1], ...]), used to generate polygon path. */
+  polygon?: number[][];
```
Comment (kristw) [inline packages/vx-voronoi/src/components/VoronoiPolygon.tsx]: `[number, number][]`

File: packages/vx-voronoi/src/components/VoronoiPolygon.tsx [hunk 2]
```diff
@@ -0,0 +1,23 @@
+import React from 'react';
+import cx from 'classnames';
+
+export type VoronoiPolygonProps = {
+  /** Override render function which is provided polygon and generated path. */
+  children?: ({ path, polygon }: { path: string; polygon: number[][] }) => React.ReactNode;
```
Comment (kristw) [inline packages/vx-voronoi/src/components/VoronoiPolygon.tsx]: `[number, number][]`

File: packages/vx-voronoi/src/components/VoronoiPolygon.tsx (hunk 1 above)
Comment (williaster) [reply packages/vx-voronoi/src/components/VoronoiPolygon.tsx]: looks like this is not possible when the type is optional, build fails with [this error](https://github.com/Microsoft/TypeScript/issues/12707#issuecomment-384878950)

Comment (williaster) [reply packages/vx-voronoi/src/components/VoronoiPolygon.tsx]: made it required to avoid

Comment (williaster) [reply packages/vx-voronoi/src/components/VoronoiPolygon.tsx]: I take that back, I think it comes from the [`@types/d3-voronoi` types](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/types/d3-voronoi/index.d.ts#L177)

**Comment by williaster** [comment]
Hrmm, not really sure how to fix this one. I think this error is coming from the [`@types/d3-voronoi` types](https://travis-ci.org/hshoff/vx/jobs/593725114#L342), and still throws even tho I ignore the call in the code 🤔

**Comment by hshoff** [comment]
Looks like the tests are failing on [`packages/vx-voronoi/test/voronoi.test.ts(24,11)`](https://github.com/hshoff/vx/blob/8502ad5f622ca77ecbb70b9072957c53d6a56c2b/packages/vx-voronoi/test/voronoi.test.ts#L24).

```js
const [, [x1, y1]] = v.extent();
```

Might try changing this to:

```js
const [[x0,y1], [x1, y1]] = v.extent();
```

**Comment by williaster** [comment]
old school syntax worked :neckbeard:  🎉

**Pull request merged**